source/server/admin/stats_render.h

```diff
@@ -0,0 +1,118 @@
+#include "envoy/stats/stats.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/server/admin/utils.h"
+
+namespace {
+constexpr uint64_t ChunkSize = 2 * 1000 * 1000;
+} // namespace
+
+namespace Envoy {
+namespace Server {
+
+// Abstract class for rendering stats. Every method is called "generate"
+// differing only by the data type, to facilitate templatized call-sites.
+//
+// There are currently Json and Text implementations of this interface, and in
+// #19546 an HTML version will be added to provide a hierarchical view.
+class StatsRender {
+public:
+  virtual ~StatsRender() = default;
+
+  // Writes a fragment for a numeric value, for counters and gauges.
+  virtual void generate(Buffer::Instance& response, const std::string& name, uint64_t value) PURE;
+
+  // Writes a json fragment for a textual value, for text readouts.
+  virtual void generate(Buffer::Instance& response, const std::string& name,
+                        const std::string& value) PURE;
+
+  // Writes a histogram value.
+  virtual void generate(Buffer::Instance& response, const std::string& name,
+                        const Stats::ParentHistogram& histogram) PURE;
+
+  // Completes rendering any buffered data.
+  virtual void render(Buffer::Instance& response) PURE;
```

Comment:
nit: would it be more appropriate to name this method `complete` or some similar things?

Reply:
Yeah that sounds good.

Reply:
actually I renamed 'finalize()'